### PR TITLE
Feat / Correct PnL calculation for derivatives

### PR DESCRIPTION
**Pull request by vic-en**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

### CODE REVIEW

**Comment by vic-en** [comment]
Result can be validated with https://docs.google.com/spreadsheets/d/14RDDA445Ic9rdoqiJ6Cm32jmLF1CrDHx3-O2tWK8EpA/edit?usp=sharing

**Comment by RC-13** [comment]
Tested and confirmed by QA, PnL calculation and return% are now taken into account. 
- confirmed that currently this is for single level, as multiple level will require more complex calculation.
Sample screenshot (bot vs spreadsheet calculator)
One closed position:


Five closed position: 

Note: Return% on the spreadsheet is a off but can be computed using the total pnl and hold portfolio

**Review by RC-13 — DISMISSED** [review]

File: hummingbot/client/performance.py [hunk 1]
```diff
@@ -53,6 +53,49 @@ def __init__(self):
         self.fees: Dict[str, Decimal] = {}
 
 
+def position_order(open: list, close: list):
```
Comment (Nullably) [inline hummingbot/client/performance.py:56]: Can you please add a return value hint.

File: hummingbot/client/performance.py [hunk 2]
```diff
@@ -53,6 +53,49 @@ def __init__(self):
...
+    for o in open:
+        for c in close:
+            if o.position == "OPEN" and c.position == "CLOSE":
+                open.remove(o)
+                close.remove(c)
+                return (o, c)
+    return None
+
+
+def aggregate_position_order(buys: list, sells: list):
```
Comment (Nullably) [inline hummingbot/client/performance.py:67]: Can you please add a return value hint.

File: hummingbot/client/performance.py [hunk 3]
```diff
@@ -53,6 +53,49 @@ def __init__(self):
...
+            aggregate_price = [s.price for s in sells if s.order_id == sell.order_id]
+            aggregate_amount = [s.amount for s in sells if s.order_id == sell.order_id]
+            sell.price = sum(aggregate_price) / len(aggregate_price)
+            sell.amount = sum(aggregate_amount)
+            aggregated_sells.append(sell)
+
+    return aggregated_buys, aggregated_sells
+
+
+def derivative_pnl(long: list, short: list):
```
Comment (Nullably) [inline hummingbot/client/performance.py:89]: Can you please add a return value hint.

File: hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py [hunk 4]
```diff
@@ -680,10 +688,10 @@ async def _status_polling_loop(self):
                 await self._poll_notifier.wait()
                 await safe_gather(
                     self._update_balances(),
-                    self._update_positions(),
-                    self._update_order_fills_from_trades(),
-                    self._update_order_status()
+                    self._update_positions()
                 )
+                await self._update_order_fills_from_trades(),
+                await self._update_order_status()
```
Comment (keithbaum) [inline hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py:694]: Why did you move out of the safe_gather these 2 methods? Shouldn't they be enforced to be grouped together in the same tick with the gather?

File: hummingbot/connector/markets_recorder.py [hunk 5]
```diff
@@ -269,10 +273,10 @@ def append_to_csv(self, trade: TradeFill):
...
         if not os.path.exists(csv_path):
             df_header = pd.DataFrame([["Config File", "Strategy", "Exchange", "Timestamp", "Market", "Base", "Quote",
-                                       "Trade", "Type", "Price", "Amount", "Fee", "Age", "Order ID", "Exchange Trade ID"]])
+                                       "Trade", "Type", "Price", "Amount", "Leverage", "Fee", "Age", "Order ID", "Exchange Trade ID", "Position"]])
             df_header.to_csv(csv_path, mode='a', header=False, index=False)
-        df = pd.DataFrame([[trade.config_file_path, trade.strategy, trade.market, trade.timestamp, trade.symbol, trade.base_asset, trade.quote_asset,
-                            trade.trade_type, trade.order_type, trade.price, trade.amount, trade.trade_fee, age, trade.order_id, trade.exchange_trade_id]])
+        df = pd.DataFrame([[trade.config_file_path, trade.strategy, trade.market, trade.timestamp, trade.symbol, trade.base_asset, trade.quote_asset, trade.trade_type,
+                            trade.order_type, trade.price, trade.amount, trade.leverage, trade.trade_fee, age, trade.order_id, trade.exchange_trade_id, trade.position]])
```
Comment (keithbaum) [inline hummingbot/connector/markets_recorder.py:279]: Watch out for latest push modifying the trade fills csv export. With the new code, it would pick up the new leverage column automatically from the TradeFill object

File: hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py (hunk 4 above)
Comment (vic-en) [reply hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py:694]: I don't want those 2 tasks to run concurrently. It'll prevent cases where orders complete event comes in without a filled event when api backup kick in.

File: hummingbot/connector/markets_recorder.py (hunk 5 above)
Comment (vic-en) [reply hummingbot/connector/markets_recorder.py:279]: Github should handle it or make me resolve conflict before merge.

File: hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py (hunk 4 above)
Comment (vic-en) [reply hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py:694]: Both tasks shouldn't ideally run concurrently.

File: hummingbot/connector/markets_recorder.py (hunk 5 above)
Comment (vic-en) [reply hummingbot/connector/markets_recorder.py:279]: Seems there's a conflict already. I'll resolve first thing today.

**Review by keithbaum — APPROVED** [review]
After discussion and explanation by @vic-en of the changes I approve. Thanks
